src/rules/noBooleanLiteralInConditionalRule.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isConditionalExpression } from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+import { needsParenthesesForNegate } from "./noBooleanLiteralCompareRule";
+
+export class Rule extends Lint.Rules.TypedRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-boolean-literal-in-conditional",
+        description: Lint.Utils.dedent`
+            Forbids either of a conditional expression's branches to be a boolean literal.
+            All such expressions can be simplified using logical operators.`,
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: ["true"],
+        type: "functionality",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING(desc: string): string {
+        return `Conditional expression with a boolean literal can be simplified to '${desc}'.`;
+    }
+
+    public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): Lint.RuleFailure[] {
+        return this.applyWithFunction(sourceFile, (ctx) => walk(ctx, program.getTypeChecker()));
+    }
+}
+
+function walk(ctx: Lint.WalkContext<void>, checker: ts.TypeChecker): void {
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+        if (isConditionalExpression(node)) {
+            const failure = check(node, checker);
+            if (failure !== undefined) {
+                ctx.addFailureAtNode(node, Rule.FAILURE_STRING(failure.desc), failure.fix);
+            }
+        }
+        return ts.forEachChild(node, cb);
+    });
+}
+
+function check(node: ts.ConditionalExpression, checker: ts.TypeChecker): { desc: string, fix: Lint.Fix | undefined } | undefined {
+    const { condition, whenTrue, whenFalse } = node;
+    const l = getLiteralBoolean(whenTrue);
+    const r = getLiteralBoolean(whenFalse);
+    // Perf: only check type of condition if there would be a failure otherwise
+    if ((l !== undefined || r !== undefined) && !isBoolean(condition, checker)) {
+        return undefined;
+    }
+
+    if (l !== undefined && r !== undefined) {
```

Comment:
IMO those conditional expressions look really nasty. It would help clarity to convert them to if ... else